### DISCUSSION TITLE
Unable to connect to MQTT broker connect ECONNREFUSED 192.168.1.2:1883

### DISCUSSION THREAD

**Discussion by Dirkystan** [tsightler/ring-mqtt#505] [Support Q&A]
I keep getting an error message econnrefused when trying to run ring-mitt on docker.
I use docker on openhabian (raspberry pi)
This is how i configured it; based on some threads i tried to play around with the mqtt url, without success.

What should i do?

docker run -it --rm --mount type=bind,source=/etc/docker,target=/data --entrypoint /app/ring-mqtt/init-ring-mqtt.js tsightler/ring-mqtt
Returned aft 2FA verification:State file /data/ring-state.json saved with updated refresh token.

docker run --rm --mount type=bind,source=/etc/docker/,target=/data tsightler/ring-mqtt

2022-11-14T12:23:18.640Z ring-mqtt Detected runmode: docker
2022-11-14T12:23:18.645Z ring-mqtt Configuration file: /data/config.json
2022-11-14T12:23:18.665Z ring-mqtt MQTT URL: mqtt://openhabian:********@192.168.1.2:1883
2022-11-14T12:23:21.919Z ring-mqtt Reading latest data from state file: /data/ring-state.json
2022-11-14T12:23:22.047Z ring-mqtt Attempting connection to Ring API using saved refresh token...
2022-11-14T12:23:22.842Z ring-mqtt Successfully established connection to Ring API using saved token
2022-11-14T12:23:22.846Z ring-mqtt Received updated refresh token
2022-11-14T12:23:22.847Z ring-mqtt Saving updated refresh token to state file
2022-11-14T12:23:23.872Z ring-mqtt Successfully saved updated state file: /data/ring-state.json
2022-11-14T12:23:24.845Z ring-mqtt Attempting connection to MQTT broker...
2022-11-14T12:23:25.019Z ring-mqtt Unable to connect to MQTT broker connect ECONNREFUSED 192.168.1.2:1883

config.json:
{
    "mqtt_url": "mqtt://127.0.0.1:1883",
    "mqtt_options": "",
    "livestream_user": "",
    "livestream_pass": "",
    "disarm_code": "",
    "enable_cameras": true,
    "enable_modes": false,
    "enable_panic": false,
    "hass_topic": "homeassistant/status",
    "ring_topic": "ring",
    "location_ids": [
        ""
    ]
}
(also MQTT URL using 192.168.1.2 ; 127.0.0.0 ; 127.0.0.1  resulted in the same error)

Docker info:

Client:
 Context:    default
 Debug Mode: false
 Plugins:
  app: Docker App (Docker Inc., v0.9.1-beta3)
  buildx: Docker Buildx (Docker Inc., v0.9.1-docker)
  compose: Docker Compose (Docker Inc., v2.12.2)

Server:
 Containers: 3
  Running: 1
  Paused: 0
  Stopped: 2
 Images: 2
 Server Version: 20.10.21
 Storage Driver: overlay2
  Backing Filesystem: extfs
  Supports d_type: true
  Native Overlay Diff: true
  userxattr: false
 Logging Driver: json-file
 Cgroup Driver: systemd
 Cgroup Version: 2
 Plugins:
  Volume: local
  Network: bridge host ipvlan macvlan null overlay
  Log: awslogs fluentd gcplogs gelf journald json-file local logentries splunk syslog
 Swarm: inactive
 Runtimes: io.containerd.runc.v2 io.containerd.runtime.v1.linux runc
 Default Runtime: runc
 Init Binary: docker-init
 containerd version: 1c90a442489720eec95342e1789ee8a5e1b9536f
 runc version: v1.1.4-0-g5fd4c4d
 init version: de40ad0
 Security Options:
  seccomp
   Profile: default
  cgroupns
 Kernel Version: 5.15.61-v8+
 Operating System: Debian GNU/Linux 11 (bullseye)
 OSType: linux
 Architecture: aarch64
 CPUs: 4
 Total Memory: 7.678GiB
 Name: openhabian
 ID: NKGT:GRNK:HRCO:O3UI:ZABW:NCKU:LYCY:OYQQ:QTT5:R5O2:MSAL:IRQG
 Docker Root Dir: /var/lib/docker
 Debug Mode: false
 Registry: https://index.docker.io/v1/
 Labels:
 Experimental: false
 Insecure Registries:
  127.0.0.0/8
 Live Restore Enabled: false

WARNING: No memory limit support
WARNING: No swap limit support

**Comment by tsightler** [comment]
MQTT URL should be seet with the information needed to connect to your MQTT broker.  I don't have any way to know what that value would be for you.
